src/coreclr/jit/morph.cpp

```diff
@@ -13826,24 +13870,49 @@ GenTree* Compiler::fgMorphModToSubMulDiv(GenTreeOp* tree)
         noway_assert(!"Illegal gtOper in fgMorphModToSubMulDiv");
     }
 
-    var_types type = tree->gtType;
+    GenTreeOp* const div = tree;
```

Comment:
Might be nice to put a before/after IR picture up in the method comment notes section.

Reply:
Done.